pkg/scheduler/algorithm/priorities/balanced_resource_allocation.go

```diff
@@ -36,21 +37,36 @@ var (
 	BalancedResourceAllocationMap = balancedResourcePriority.PriorityMap
 )
 
-func balancedResourceScorer(requested, allocable *schedulercache.Resource) int64 {
+func balancedResourceScorer(requested, allocable *schedulercache.Resource, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
```

Comment:
is it possible to also merge `volumes` into `Resource`? so no interface changed.

Reply:
I thought about it but it comes mostly under scalar resource and I think there is no definitive way on kubelet side to introspect node and get that information but it would be ideal to have that kind of information there. @bsalamat WDYT?

Reply:
> is it possible to also merge volumes into Resource? so no interface changed.

That would require updating the scheduler cache with the volume information and that in turn would require Kubelets to report volume counts, etc. as a part of allocatable resources. This would be a larger project requiring involvement of the node team. I won't go this path, if we can solve the problem without changes to the kubelet.

Reply:
@gnufied is working on a design for dynamic node reporting of attachable volume limits per plugin.  So I think when that is implemented, the limits will actually will be in the Node object.

Reply:
> I think when that is implemented, the limits will actually will be in the Node object.

That will be great and will make our logic cleaner.

Reply:
> So I think when that is implemented, the limits will actually will be in the Node object.

Thanks @msau42 . That will be awesome. Ohh btw, then we could actually cache in nodeinfo :).

Reply:
That may related to our previous comments: did not update nodeInfo outside of cache :)

>> I think when that is implemented, the limits will actually will be in the Node object.

> That will be great and will make our logic cleaner.

Great, @gnufied any related doc/issue?

Reply:
@k82cn @bsalamat https://github.com/kubernetes/community/pull/2051